### PR TITLE
feat: adding arbitrary data field for market contracts

### PR DESCRIPTION
**Pull request by ramenforbreakfast**

### CODE REVIEW

File: apps/smart-contracts/core/test/fixtures/PrePOMarketFixture.ts [hunk 1]
```diff
@@ -43,8 +44,8 @@ export async function smockPrePOMarketFixture(
     marketParams.floorValuation,
     marketParams.ceilingValuation,
     marketParams.expiryTime,
-    marketParams.hookBeacon,
-    marketParams.feeBeacon
+    marketParams.hookBeacon ? marketParams.hookBeacon : ZERO_ADDRESS,
+    marketParams.feeBeacon ? marketParams.feeBeacon : ZERO_ADDRESS
```
Comment (XavierEkkel) [inline apps/smart-contracts/core/test/fixtures/PrePOMarketFixture.ts:48]: this is unrelated to the new data field right?

**Review by XavierEkkel — APPROVED** [review]
LGTM

File: apps/smart-contracts/core/test/fixtures/PrePOMarketFixture.ts (hunk 1 above)
Comment (ramenforbreakfast) [reply apps/smart-contracts/core/test/fixtures/PrePOMarketFixture.ts:48]: Yes sorry I should have included in this PR notes regarding that some of these changes are related to the fact that integration tests were not being run on accident and thus I did not catch this breaking change in our fixtures related to the introduction of fee beacons.
